File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/spell/focused_fury/FocusedFuryData.java
package aurocosh.divinefavor.common.custom_data.player.data.spell.focused_fury;

import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityList;
import net.minecraft.util.ResourceLocation;

// The default implementation of the capability. Holds all the logic.
public class FocusedFuryData {
    private ResourceLocation mobTypeId;

    public FocusedFuryData() {
        this.mobTypeId = new ResourceLocation("");
    }

    public ResourceLocation getMobTypeId() {
        return mobTypeId;
    }

    public void reset(){
        mobTypeId = new ResourceLocation("");
    }

    public void setMobTypeId(ResourceLocation typeId) {
        mobTypeId = typeId;
    }

    public void setFury(Entity entity) {
        mobTypeId = EntityList.getKey(entity);
    }

    public boolean hasFury(Entity entity) {
        return EntityList.getKey(entity).equals(mobTypeId);
    }

    public boolean hasFury() {
        return !mobTypeId.getPath().isEmpty();
    }

    public String getMobName() {
        String translationName = EntityList.getTranslationName(mobTypeId);
        return translationName == null ? "divinefavor:no_fury_assigned" : translationName;
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/player/data/spell/focused_fury/FocusedFuryData.kt
package aurocosh.divinefavor.common.custom_data.player.data.spell.focused_fury

import aurocosh.divinefavor.common.lib.EmptyConst.emptyLocation
import net.minecraft.entity.Entity
import net.minecraft.entity.EntityList
import net.minecraft.util.ResourceLocation

// The default implementation of the capability. Holds all the logic.
class FocusedFuryData {
    var mobTypeId: ResourceLocation = emptyLocation()

    val mobName: String
        get() {
            val translationName = EntityList.getTranslationName(mobTypeId)
            return translationName ?: "divinefavor:no_fury_assigned"
        }

    fun reset() {
        mobTypeId = emptyLocation()
    }

    fun setFury(entity: Entity) {
        mobTypeId = EntityList.getKey(entity) ?: emptyLocation()
    }

    fun hasFury(entity: Entity): Boolean {
        return EntityList.getKey(entity) == mobTypeId
    }

    fun hasFury(): Boolean {
        return mobTypeId.path.isNotEmpty()
    }
}
